Some of the "command" package tests modify their fixtures in-place

The intended model for our `command` package tests is for them to either consider their inputs read-only or to copy the fixtures to a temporary directory so that the filesystem can be modified without impacting other tests.

However, we seem to have a few tests for `terraform plan` that are modifying their fixture directories in-place, causing leftover extra files after the test runs complete and presumably also causing the tests sharing a single fixture directory to affect one another's behaviors and also potentially for one run of a test to affect a subsequent run of it in the same work tree.

I tracked down three examples of this by temporarily making the `plan` test fixture read-only, since that seems to be the one where `terraform.tfstate` files seem to show up after a `command` package test run:

```
--- FAIL: TestPlan (0.00s)
    plan_test.go:48: bad: 1
        
        
        Error: Error locking state: Error acquiring the state lock: open terraform.tfstate: permission denied
        
        Terraform acquires a state lock to protect the state from being written
        by multiple users at the same time. Please resolve the issue above and try
        again. For most commands, you can disable locking with the "-lock=false"
        flag, but this is not recommended.
        
        
--- FAIL: TestPlan_lockedState (3.29s)
    plan_test.go:61: statelocker wrote: open /home/mart/Devel/terraform/terraform/command/testdata/plan/terraform.tfstate: permission denied
2020/04/03 10:06:37 [DEBUG] GET http://127.0.0.1:39247
--- FAIL: TestPlan_detailedExitcode (0.00s)
    plan_test.go:749: bad: 1
        
        
        Error: Error locking state: Error acquiring the state lock: open terraform.tfstate: permission denied
        
        Terraform acquires a state lock to protect the state from being written
        by multiple users at the same time. Please resolve the issue above and try
        again. For most commands, you can disable locking with the "-lock=false"
        flag, but this is not recommended.
```

It would be good to rework these tests to use the pattern of copying the fixture into a temporary directory before doing any operations.

--- a/command/command_test.go
+++ b/command/command_test.go
@@ -540,7 +540,7 @@ func testChdir(t *testing.T, new string) func() {
 }
 
 // testCwd is used to change the current working directory
-// into a test directory that should be remoted after
+// into a test directory that should be removed after
 func testCwd(t *testing.T) (string, string) {
 	t.Helper()
 

--- a/command/plan_test.go
+++ b/command/plan_test.go
@@ -25,14 +25,10 @@ import (
 )
 
 func TestPlan(t *testing.T) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	if err := os.Chdir(testFixturePath("plan")); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
 	p := planFixtureProvider()
 	ui := new(cli.MockUi)
@@ -50,22 +46,16 @@ func TestPlan(t *testing.T) {
 }
 
 func TestPlan_lockedState(t *testing.T) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
-	testPath := testFixturePath("plan")
-	unlock, err := testLockState(testDataDir, filepath.Join(testPath, DefaultStateFilename))
+	unlock, err := testLockState(testDataDir, filepath.Join(td, DefaultStateFilename))
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer unlock()
-
-	if err := os.Chdir(testPath); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
 
 	p := planFixtureProvider()
 	ui := new(cli.MockUi)
@@ -489,14 +479,10 @@ func TestPlan_validate(t *testing.T) {
 	test = false
 	defer func() { test = true }()
 
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	if err := os.Chdir(testFixturePath("plan-invalid")); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan-invalid"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
 	p := testProvider()
 	p.GetSchemaReturn = &terraform.ProviderSchema{
@@ -726,14 +712,10 @@ func TestPlan_varFileWithDecls(t *testing.T) {
 }
 
 func TestPlan_detailedExitcode(t *testing.T) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	if err := os.Chdir(testFixturePath("plan")); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
 	p := planFixtureProvider()
 	ui := new(cli.MockUi)
@@ -751,14 +733,10 @@ func TestPlan_detailedExitcode(t *testing.T) {
 }
 
 func TestPlan_detailedExitcode_emptyDiff(t *testing.T) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	if err := os.Chdir(testFixturePath("plan-emptydiff")); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan-emptydiff"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
 	p := testProvider()
 	ui := new(cli.MockUi)
@@ -855,14 +833,10 @@ func TestPlan_shutdown(t *testing.T) {
 }
 
 func TestPlan_init_required(t *testing.T) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	if err := os.Chdir(testFixturePath("plan")); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-	defer os.Chdir(cwd)
+	td := tempDir(t)
+	copy.CopyDir(testFixturePath("plan"), td)
+	defer os.RemoveAll(td)
+	defer testChdir(t, td)()
 
 	ui := new(cli.MockUi)
 	c := &PlanCommand{

